pkg/ddl/ddl_api.go

```diff
@@ -7252,7 +7256,7 @@ func (d *ddl) createIndex(ctx sessionctx.Context, ti ast.Ident, keyType ast.Inde
 				"a background job is trying to add the same index, "+
 				"please check by `ADMIN SHOW DDL JOBS`", indexName)
 		} else {
-			err = dbterror.ErrDupKeyName.GenWithStack("index already exist %s", indexName)
+			err = dbterror.ErrDupKeyName.GenWithStackByArgs(indexName)
```

Comment:
Do we need to update line 7255 like this(using `Duplicate key name` instead of `index already exist`)?
Besides, do we need to unify this error msg in https://github.com/pingcap/tidb/blob/3682bd82cac05a75241ed9fc9f29a1d320d776a7/pkg/ddl/index.go#L567

Reply:
Yes this is needed to address https://github.com/pingcap/tidb/issues/48040#issuecomment-1783773545

Looks like this is used in three locations:
```
$ git grep '"index already exist' | grep -v '_test.go'
pkg/ddl/ddl_api.go:			err = dbterror.ErrDupKeyName.GenWithStack("index already exist %s; "+
pkg/ddl/index.go:			err = dbterror.ErrDupKeyName.GenWithStack("index already exist %s", indexName)
pkg/ddl/schematracker/dm_tracker.go:		return dbterror.ErrDupKeyName.GenWithStack("index already exist %s", indexName)
```

If might be useful to see what situations trigger those and if this is compatible with MySQL.

Reply:
For compatibility with MySQL we should make sure that:
- The error code is the same unless this is really impossible.
- The error text is the same unless there is a strong reason for deviating. If we deviate we should try to add to the message instead of replacing it.

Reply:
For the error in `pkg/ddl/ddl_api.go`:

This seems to happen when a background job is adding the same index. I think this is a TiDB specific situation, so this is probably fine.

Reply:
I think it should be changed to what I said about `pkg/ddl/ddl_api.go`, leaving the supplementary information intact, but replacing `index already exist` with `Duplicate key name`. 
Because I think line7255 and line7259 are the same error, but line7255 adds some extra information.

Reply:
done in 2e73d713dc900caf11a8fe4be30155feb6b33ae2